### DISCUSSION TITLE
[1.6] 2022-05-23

### DISCUSSION THREAD

**Discussion by darthcloud** [darthcloud/BlueRetro#285] [General]
# General
* Add button macro to perform system function (alternative to boot button)
* Add support for wireless ctrl reassignment base on wired ctrl connection (Internal install).
* Add support for powering on/off and reset (Internal install)
* Add BLE commands to reset ESP32
* Add BLE attribute to get BD_ADDR (Closes #249)
* Add macro and BLE commands to put ESP32 in deep sleep (Closes #174)
* Detect ESP32 PICO package and use GPIO20 for LED (Closes #156)
* Make BLE advertised name unique

# Wireless
* Add support for 8bitdo N64 controller
* Add rumble feedback on accessory switch (Closes #175)

# Wired
* [N64] Switch data line config from open drain to push-pull (Closes #264)
* [WII-EXT] Add support for Wii extension up to 2 players (tested SNES mini only, Hakchi kernel not working)

<hr /><em>This discussion was created from the release <a href='https://github.com/darthcloud/BlueRetro/releases/tag/v1.6'>[1.6] 2022-05-23</a>.</em>

**Comment by khtp** [comment]
when you add the support of xbox controls for the bluethooht BLE so that the xbox one controller FW: 5.3 on ps2 does not disconnect? is it the psx-ex or is it still missing?

**Reply by darthcloud** [reply]
You said last month that your issue was resolved:
https://github.com/darthcloud/BlueRetro/issues/237#issuecomment-1105738509

**Reply by khtp** [reply]
the problem about it detecting me on the ps2 fat as slim already works without problem I was missing pin 4 which is GND connect it to the esp32 of my ps2 controller cable, the problem now is that when you play in the middle of the game the controller is disconnected, I have to restart pairing mode to get the controller to work again and it repeats like 25 minutes.

**Reply by khtp** [reply]
in the game it says: connect your ps2 control because it disconnected, connect it again, it happens every hour or every 30 minutes or when I leave it paused for more than 30 minutes, on other consoles the same thing happens to me when I use the xbox one s control 2 generation FW: 5.3 on wii the same thing happens to me, beware it is only when I use the xbox controller with other controllers such as wii u pro controller those problems do not happen to me.

**Reply by khtp** [reply]
I have fw 5.3 on my xbox one s controller, the following happens to me: when I'm playing out of nowhere the console stops recognizing the controller and a message appears that says: disconnect and connect player 1, what I do is turn off and turn on the esp32 again so that the console detects me again that I have the controller connected and the esp32 matches the xbox one s controller and the console detects it again, in short, that happens, I only have 1 controller. It happens to me every 30 minutes or when I leave the game paused for a long time.

**Reply by khtp** [reply]
Is there any way to send you a test example demonstrating what happens to me? I sent you a log or a video?

**Reply by darthcloud** [reply]
Not really, I just need to have some time to actually play a games as this is not instantly reproducible.

 I updated my controller FW to 5.13.3143.0 today and will try to play and see this weekend.

**Comment by ghost** [comment]
I was hoping you could clarify something for me. Does the PICO detection only swap the status led to pin 20? 

I was looking to pick up some ESP32-PICO-KIT devboards which have access to some GPIO pins that other PICO boards are lacking.

Looking at the board layout again, it looks like the ESP32-PICO-KIT doesn't have an IO20. Would it cause issues not having that pin?

**Reply by darthcloud** [reply]
Only the new PICO_V3 (not D4) is supported.

And it will will work only for n64, gc and PlayStation.

For those system the only pin change is the LED that move to IO20

**Reply by ghost** [reply]
Ok, thanks for the clarification.
